src/lib/web-component/log-view/log-view-element.js

```diff
@@ -51,6 +95,35 @@ export class LogView extends HTMLElement {
     this.tableBody.appendChild(rowsFragment);
   }
 
+  handleEvent(event) {
+    if (event.type === 'click') {
+      const logDetails = event.target.closest('tr')?._log;
+
+      if (logDetails) {
+        this.openDetailSidebar(logDetails);
+        return;
+      }
+
+      switch (event.target) {
+        case this.closeBtn:
+          this.closeDetailSidebar();
+          break;
+        case this.clearTSFilterBtn:
+          this.hideTSFilter();
+          break;
+        default:
+          throw new Error(`wrong event target found: ${event.target.tagName}`);
+      }
+    } else if (event.type === 'contextmenu') {
+      this.selectedRow = event.target.closest('tr');
+      if (this.selectedRow) {
+        browser.menus.overrideContext({});
+        this.createContextMenu();
+        browser.menus.onClicked.addListener(this.menuOnCickListener);
```

Comment:
Register the menus and events OUTSIDE the 'contextmenu' click handler, because the menu items themselves are static and don't change.

In the `menus.onClicked` listener, you can use the `browser.menus.getTargetElement` method to get the element that was clicked. This can be useful when the menu was created from multiple documents (because then your script can ignore `menus.onClicked` calls that originate from another document).